include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1230 @@
+     *
+     * @return bool Success/Failure
+     */
+    public function pushEvent($event_local, $event_remote = NULL)
+    {
+
+        if (!isset($event_remote) || empty($event_remote))
+        {
+            $event = $this->createGoogleCalendarEvent($event_local);
+            $return = $this->gService->events->insert($this->calendarId, $event);
+        }
+        else {
+            $event = $this->updateGoogleCalendarEvent($event_local, $event_remote);
+            $return = $this->gService->events->update($this->calendarId, $event->getId(), $event);
+        }
+
+        // Set the SuiteCRM Meeting's last sync timestamp, and google id
+        $this->setLastSync($event_local, $return->getId());
+
+        // We don't get a status code back showing success. Instead, the return of the create or update is the Google_Service_Calendar_Event object after saving. So we check to make sure it has an ID to determine Success/Failure.
+        if (isset($return->id))
+        {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * Pull event from Google Calendar to SuiteCRM
+     *
+     * If the SuiteCRM Meeting is not provided, a new one will be created
+     * and inserted. If one is provided, the existing meeting will be updated.
+     *
+     * @param Google_Service_Calendar_Event $event_remote Tthe Google event
+     * @param Meeting $event_local Meeting (optional) SuiteCRM meeting bean
+     *
+     * @return bool Success/Failure of setLastSync, since that's what saves the record
+     */
+    public function pullEvent($event_remote, $event_local = NULL)
+    {
+
+        if (!isset($event_local) || empty($event_local))
+        {
+            $event = $this->createSuitecrmMeetingEvent($event_remote);
+        }
+        else {
+            $event = $this->updateSuitecrmMeetingEvent($event_local, $event_remote);
+        }
+
+        // We need to set the suitecrm_ private properties in the Google event here,
+        // Otherwise it's seen as a new event next time we sync
+
+        // We pull the existing extendedProperties, and change our values
+        // That way we don't mess with anything else that's using other values.
+        $extendedProperties = $event_remote->getExtendedProperties();
+
+        if (!empty($extendedProperties))
+        {
+            $private = $extendedProperties->getPrivate();
+        }
+        else {
+            $extendedProperties = new Google_Service_Calendar_EventExtendedProperties;
+            $private = array();
+        }
+
+        $private['suitecrm_id'] = $event->id;
+        $private['suitecrm_type'] = $event->module_name;
+
+        $extendedProperties->setPrivate($private);
+
+        $event_remote->setExtendedProperties($extendedProperties);
+
+        $greturn = $this->gService->events->update($this->calendarId, $event_remote->getId(), $event_remote);
```

Comment:
Variable appears to unused. Please consider remove the variable or adding a test an throwing and exception

Reply:
I was using it for debugging, but I should be checking it. I'll add the check.

Reply:
Check added.